src/base/system_parameter.c

```diff
@@ -7387,23 +7545,17 @@ sysprm_reload_and_init (const char *db_name, const char *conf_file)
  */
 static int
 prm_load_by_section (INI_TABLE * ini, const char *section, bool ignore_section, bool reload, const char *file,
-		     const int load_flags, bool ignore_case)
+		     const int load_flags, bool ignore_case, bool is_common_section)
 {
   int i, error;
   int sec_len;
   const char *sec_p;
   const char *key, *value;
   SYSPRM_PARAM *prm;
-  bool on_server = false;
-  bool on_client = false;
-
-#if defined(SERVER_MODE)
-  on_server = true;
-#endif
+  int sec_key_len;
 
-#if defined(CS_MODE)
-  on_client = true;
-#endif
+  sec_p = strchr (section, ':');
+  sec_key_len = (int) ((sec_p) ? CAST_STRLEN (sec_p - section) : strlen (section));
```

Comment:
```suggestion
  sec_key_len = (int) ((sec_p) ? CAST_STRLEN (sec_p - section) : strlen (section));
```
```suggestion
  sec_key_len = (sec_p) ? CAST_STRLEN (sec_p - section) : strlen (section);
```
이미 캐스팅되어 있어서 추가적인 캐스팅이 필요하지 않을을 거 같습니다.

Reply:
strlen()의 결과도 casting 대상이 되어야 합니다.